pkg/config/application.go

```diff
@@ -396,6 +402,12 @@ type WaitApprovalStageOptions struct {
 	MinApproverNum int      `json:"minApproverNum" default:"1"`
 }
 
+type CustomSyncOptions struct {
+	Timeout Duration          `json:"timeout"`
```

Comment:
should use default to set default value for this option
ref: https://github.com/pipe-cd/pipecd/blob/master/pkg/config/piped.go#L56

Reply:
https://github.com/pipe-cd/pipecd/blob/master/pkg/config/application.go/#L396
WaitApproval timeout define default time  not in the structure but in the unmarshaljson.
https://github.com/pipe-cd/pipecd/blob/master/pkg/config/application.go/#L268

Reply:
Which do you think is better? Set in executor side or wrap all in configuration stub 🤔

Reply:
I think the latter one (set it in the stub) is better in terms of the visibility.

Reply:
Thank you for this PR.
https://github.com/pipe-cd/pipecd/pull/4258/files

Reply:
Please add validate function for this stage as I mentioned in https://github.com/pipe-cd/pipecd/pull/4223#discussion_r1128940960